src/App.jsx

```diff
@@ -14,45 +16,90 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_ALPHABET = 'byAlphabet';
+const SORT_FIELD_LENGTH = 'byLength';
+
+function preparedGoodsFromServer(goods, sortField, reverse) {
+  const preperedGoods = [...goods];
+
+  if (sortField) {
+    preperedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_ALPHABET:
+          return good1.localeCompare(good2);
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    preperedGoods.reverse();
+  }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
+  return preperedGoods;
+}
+
+const buttonClass = (type, sortField, sortBy) => (
+  `button ${type} ${sortField !== sortBy ? 'is-light' : ''}`
 );
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReverse, setReverse] = useState(false);
+
+  const forVisibleReset = sortField || isReverse;
```

Comment:
please try to use more meaningful names
```suggestion
  const isShowResetButton = sortField || isReverse;
```